src/openrct2-ui/interface/Window.cpp

```diff
@@ -746,7 +746,44 @@ void Window::DrawWidgets(rct_drawpixelinfo& dpi)
 
 void Window::Close()
 {
-    window_close(*this);
+    CloseWindowModifier modifier = closedWithModifier();
+
+    if (modifier == CloseWindowModifier::CLOSE_WINDOW_MODIFIER_SHIFT)
+    {
+        CloseOthers();
+    }
+    else if (modifier == CloseWindowModifier::CLOSE_WINDOW_MODIFIER_CONTROL)
+    {
+        CloseOthersOfThisClass();
+    }
+    else
+    {
+        window_close(*this);
+    }
+}
+
+void Window::CloseOthers()
+{
+    window_close_all_except_number_and_class(number, classification);
+}
+
+void Window::CloseOthersOfThisClass()
+{
+    window_close_by_class(classification);
+}
+
+CloseWindowModifier Window::closedWithModifier()
```

Comment:
Methods should be title case and start with a verb in present tense. E.g. `GetCloseModifier()`.